substrate-archive-backend/src/read_only_backend/blockchain_backend.rs

```diff
@@ -94,13 +99,13 @@ impl<Block: BlockT> BlockchainBackend<Block> for ReadOnlyBackend<Block> {
 
 impl<Block: BlockT> HeaderBackend<Block> for ReadOnlyBackend<Block> {
     fn header(&self, id: BlockId<Block>) -> ChainResult<Option<Block::Header>> {
-        util::read_header::<Block>(&self.db, columns::KEY_LOOKUP, columns::HEADER, id)
+        util::read_header::<Block>(self.db.clone(), columns::KEY_LOOKUP, columns::HEADER, id)
             .map_err(|e| BlockchainError::Msg(e.to_string()))
     }
 
     fn info(&self) -> Info<Block> {
         // TODO: Remove expect
-        let meta = util::read_meta::<Block>(&self.db, columns::HEADER)
+        let meta = util::read_meta::<Block>(self.db.clone(), columns::HEADER)
```

Comment:
if you agree with changing: 
```rust
pub fn read_meta<Block>(
     db: &dyn ReadOnlyDatabaseTrait,
     col_header: u32,
 ) 
```

```suggestion
        let meta = util::read_meta::<Block>(&*self.db, columns::HEADER)
```